Upgraded aws-sdk-go dependency to v1.35.5 to support new regions like Milan

jqmichael opened the pull request:
**What type of PR is this?**
/kind feature

**What this PR does / why we need it**:
Upgraded aws-sdk-go to v1.35.5 to include new regions, i.e. Millan

**Which issue(s) this PR fixes**:

Fixes #
NONE

**Special notes for your reviewer**:
Commands to upgrade the version.
```
./hack/pin-dependency.sh github.com/aws/aws-sdk-go v1.35.5
./hack/update-vendor.sh
./hack/lint-dependencies.sh

// Additional changes required by lint-dependencies
hack/pin-dependency.sh github.com/jmespath/go-jmespath v0.4.0
./hack/update-vendor.sh
./hack/lint-dependencies.sh
```

**Does this PR introduce a user-facing change?**:
NONE

**Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.**:
NONE

```release-note
NONE
```

mogren commented on go.mod:
```diff
@@ -19,7 +19,7 @@ require (
 	github.com/armon/circbuf v0.0.0-20150827004946-bbbad097214e
 	github.com/auth0/go-jwt-middleware v0.0.0-20170425171159-5493cabe49f7 // indirect
-	github.com/aws/aws-sdk-go v1.28.2
+	github.com/aws/aws-sdk-go v1.31.6
```
The actual change

wongma7 commented:
/ok-to-test

mogren reviewed (approved):
/lgtm

wongma7 commented:
/test pull-kubernetes-e2e-kops-aws

wongma7 commented:
/lgtm

wongma7 commented:
/lgtm cancel

wongma7 commented:
all the BUILD changes pointing to a client-go vendor directory (???) are definitely unrelated

jqmichael commented:
/test pull-kubernetes-e2e-kops-aws

jqmichael commented:
/retest

jqmichael commented:
```
W0528 00:40:39.584] ERROR: /go/src/k8s.io/kubernetes/vendor/BUILD:8:1: no such package 'vendor/github.com/aws/aws-sdk-go/internal/sync/singleflight': BUILD file not found in any of the following directories. Add a BUILD file to a directory to mark it as a package.
```
Looks like test-infra dependency needs to be upgraded first. 

https://github.com/kubernetes/test-infra/blob/master/go.mod#L29

cheftako commented:
/remove-sig api-machinery

jqmichael commented:
> ```
> W0528 00:40:39.584] ERROR: /go/src/k8s.io/kubernetes/vendor/BUILD:8:1: no such package 'vendor/github.com/aws/aws-sdk-go/internal/sync/singleflight': BUILD file not found in any of the following directories. Add a BUILD file to a directory to mark it as a package.
> ```
> 
> Looks like test-infra dependency needs to be upgraded first.
> 
> https://github.com/kubernetes/test-infra/blob/master/go.mod#L29

Finally figured it out (had to go through [bazel 101](https://docs.bazel.build/versions/3.2.0/build-ref.html#packages)). 

The `BUILD` file was not uploaded to `vendor/github.com/aws/aws-sdk-go/internal/sync/singleflight` because I defined in my global `gitignore` policy to remove `build` target (meant to ignore `/build/` directory). 

Check ignored files by git 
```
git status --ignored 
```

Find out why a particular file was ignored by git 
```
git check-ignore --verbose --non-matching <file_path>
```

jqmichael commented:
/test pull-kubernetes-e2e-kops-aws

jqmichael commented:
/test pull-kubernetes-e2e-kops-aws

jqmichael commented:
/retest

jqmichael commented:
/retest

jqmichael commented:
`pull-kubernetes-e2e-kops-aws` is failing because kops doesn't support 1.19 yet. 

```
This version of kubernetes is not yet supported; upgrading kops is required
```
```
	KOPS_RELEASE_VERSION = "1.18.0-alpha.3"
```
https://github.com/kubernetes/kops/blob/master/version.go#L26

Looking into how I can help get kops 1.19 roll out.

jqmichael commented:
/test pull-kubernetes-e2e-kops-aws

jqmichael commented:
/retest

jqmichael commented:
/retest

jqmichael commented:
```
W0708 13:12:49.909] cannot determine hash for "https://storage.googleapis.com/kubernetes-release-pull/ci/pull-kubernetes-e2e-kops-aws/v1.19.0-beta.2.793+01d6753b5c7271/bin/linux/arm64/kubelet" (have you specified a valid file location?)

W0708 13:12:49.939] 2020/07/08 13:12:49 process.go:155: Step '/workspace/kops create cluster --name e2e-bd9ab6df6f-dba53.test-cncf-aws.k8s.io --ssh-public-key /workspace/.ssh/kube_aws_rsa.pub --node-count 4 --node-volume-size 48 --master-volume-size 48 --master-count 1 --zones sa-east-1c --master-size c5.large --kubernetes-version https://storage.googleapis.com/kubernetes-release-pull/ci/pull-kubernetes-e2e-kops-aws/v1.19.0-beta.2.793+01d6753b5c7271 --admin-access 35.226.123.140/32 --cloud aws --override cluster.spec.nodePortAccess=0.0.0.0/0 --yes' finished in 9.795259329s

W0708 13:12:49.945] 2020/07/08 13:12:49 process.go:153: Running: kubectl -n kube-system get pods -ojson -l k8s-app=kops-controller

W0708 13:12:50.739] The connection to the server localhost:8080 was refused - did you specify the right host or port?

W0708 13:12:50.759] 2020/07/08 13:12:50 kubernetes.go:117: kubectl get pods failed: error during kubectl -n kube-system get pods -ojson -l k8s-app=kops-controller: exit status 1

W0708 13:12:50.802] W0708 13:12:50.802076    7513 vfs_castore.go:604] CA private key was not found
W0708 13:12:50.803] 
W0708 13:12:50.803] cannot find CA certificate
```

https://prow.k8s.io/view/gcs/kubernetes-jenkins/pr-logs/pull/91513/pull-kubernetes-e2e-kops-aws/1280849041231450112/

jqmichael commented:
```
			"containerStatuses": [
				{
					"name": "test",
					"state": {
						"terminated": {
							"exitCode": 1,
							"reason": "Error",
							"message": "ication/json -h x-goog-if-generation-match:1594176687871906 cp /tmp/gsutil_EGCkk7 gs://kubernetes-jenkins/pr-logs/directory/pull-kubernetes-e2e-kops-aws/jobResultsCache.json

I0708 13:13:14.617] process 8548 exited with code 0 after 0.0m
I0708 13:13:14.617] Call:  gsutil stat gs://kubernetes-jenkins/pr-logs/pull/91513/pull-kubernetes-e2e-kops-aws/jobResultsCache.json
I0708 13:13:16.997] process 8728 exited with code 0 after 0.0m
I0708 13:13:16.999] Call:  gsutil -q cat 'gs://kubernetes-jenkins/pr-logs/pull/91513/pull-kubernetes-e2e-kops-aws/jobResultsCache.json#1594176693230302'
I0708 13:13:19.526] process 8872 exited with code 0 after 0.0m
I0708 13:13:19.528] Call:  gsutil -q -h Content-Type:application/json -h x-goog-if-generation-match:1594176693230302 cp /tmp/gsutil__jqiOb gs://kubernetes-jenkins/pr-logs/pull/91513/pull-kubernetes-e2e-kops-aws/jobResultsCache.json
I0708 13:13:22.895] process 9018 exited with code 0 after 0.1m
I0708 13:13:22.896] Call:  gsutil -q -h Content-Type:application/json cp /tmp/gsutil_UFvNUR gs://kubernetes-jenkins/pr-logs/pull/91513/pull-kubernetes-e2e-kops-aws/1280849041231450112/finished.json
I0708 13:13:26.292] process 9198 exited with code 0 after 0.1m
I0708 13:13:26.293] Call:  gsutil -q -h Content-Type:text/plain -h 'Cache-Control:private, max-age=0, no-transform' cp /tmp/gsutil_zWMlkv gs://kubernetes-jenkins/pr-logs/directory/pull-kubernetes-e2e-kops-aws/latest-build.txt
I0708 13:13:29.467] process 9378 exited with code 0 after 0.1m
I0708 13:13:29.468] Call:  gsutil -q -h Content-Type:text/plain -h 'Cache-Control:private, max-age=0, no-transform' cp /tmp/gsutil_Wq4SZY gs://kubernetes-jenkins/pr-logs/pull/91513/pull-kubernetes-e2e-kops-aws/latest-build.txt
I0708 13:13:32.681] process 9558 exited with code 0 after 0.1m
I0708 13:13:32.681] Call:  gsutil -q cp /workspace/build-log.txt gs://kubernetes-jenkins/pr-logs/pull/91513/pull-kubernetes-e2e-kops-aws/1280849041231450112/build-log.txt
I0708 13:13:35.531] process 9738 exited with code 0 after 0.0m
+ EXIT_VALUE=1
+ set +o xtrace
",
```

https://storage.googleapis.com/kubernetes-jenkins/pr-logs/pull/91513/pull-kubernetes-e2e-kops-aws/1280849041231450112/podinfo.json

hakman commented:
/retest

hakman commented:
/retest

hakman commented:
/retest

jqmichael commented:
/retest

hakman commented:
This is required: https://github.com/kubernetes/kubernetes/pull/91513#issuecomment-656259590

Also, maybe you want latest if will rerun all tests anyway:
https://github.com/aws/aws-sdk-go/releases/tag/v1.33.5

neolit123 commented:
/remove-sig cluster-lifecycle

hakman commented:
@neolit123 If needed, I can create a new PR with latest version.

neolit123 commented:
hi @hakman
>  If needed, I can create a new PR with latest version.

not sure what the desired action is here. i was just removing the SIG CL label.

jqmichael commented:
/test pull-kubernetes-e2e-kops-aws

jqmichael commented:
/assign @liggitt @dims

liggitt commented:
this looks identical to #94909, and would be blocked on the same issue - https://github.com/kubernetes/kubernetes/pull/94909#discussion_r491445429

jqmichael committed:
Upgraded aws-sdk-go to v1.35.5 to include more regions, i.e. Millan

./hack/pin-dependency.sh github.com/aws/aws-sdk-go v1.35.5
./hack/update-vendor.sh
./hack/lint-dependencies.sh

// Additional changes required by lint-dependencies
hack/pin-dependency.sh github.com/jmespath/go-jmespath v0.4.0
./hack/update-vendor.sh
./hack/lint-dependencies.sh

jqmichael commented:
/test pull-kubernetes-e2e-kops-aws

jqmichael commented:
/retest

hakman commented:
Failed kops test is unrelated to the fix. It is only a string mismatch due to https://github.com/kubernetes/kubernetes/pull/95202 and a rebase may fix it, but don't know if worth the trouble.
/lgtm

jqmichael commented:
> It is only a string mismatch due to #95202 and a rebase may fix it

Thanks @hakman. My branch already contains changes in #95202. Any idea where the discrepancy in kubectl is? Is kubectl running in the test built from my branch or from an older commit?  

Thanks

hakman commented:
@jqmichael Sorry, but can't find where the difference is coming from.

liggitt commented:
/lgtm
/approve
/skip

jqmichael commented:
/retest

jqmichael commented:
/retest

jqmichael commented:
/test pull-kubernetes-e2e-kops-aws
